src/Sulu/Bundle/WebsiteBundle/Routing/ContentRouteProvider.php

```diff
@@ -122,27 +131,22 @@ public function getRouteCollectionForRequest(Request $request)
         $htmlExtension = '.html';
         $resourceLocator = $this->requestAnalyzer->getResourceLocator();
 
-        if (
-            $this->requestAnalyzer->getMatchType() == RequestAnalyzerInterface::MATCH_TYPE_REDIRECT
+        if ($this->requestAnalyzer->getMatchType() == RequestAnalyzerInterface::MATCH_TYPE_REDIRECT
             || $this->requestAnalyzer->getMatchType() == RequestAnalyzerInterface::MATCH_TYPE_PARTIAL
         ) {
-            // redirect webspace correctly with locale
-            $collection->add(
-                'redirect_' . uniqid(),
-                $this->getRedirectWebSpaceRoute()
-            );
-        } elseif (
-            $request->getRequestFormat() === 'html' &&
-            substr($request->getPathInfo(), -strlen($htmlExtension)) === $htmlExtension
+            $url = $this->getRedirectWebSpaceUrl();
+            if ($this->matchRedirectUrl($url)) {
+                // redirect webspace correctly with locale
+                $collection->add('redirect_' . uniqid(), $this->getRedirectWebSpaceRoute($url));
+            }
+        } elseif ($request->getRequestFormat() === 'html'
+                  && substr($request->getPathInfo(), -strlen($htmlExtension)) === $htmlExtension
         ) {
-            // redirect *.html to * (without url)
-            $collection->add(
-                'redirect_' . uniqid(),
-                $this->getRedirectRoute(
-                    $request,
-                    $this->getUrlWithoutEndingTrailingSlash($resourceLocator)
-                )
-            );
+            $url = $this->getUrlWithoutEndingTrailingSlash($resourceLocator);
+            if ($this->matchRedirectUrl($url)) {
+                // redirect *.html to * (without url)
+                $collection->add('redirect_' . uniqid(), $this->getRedirectRoute($request, $url));
```

Comment:
could we use the $url instead of the uniqid? would be better for tools like new relic